pkg/agent/run.go

```diff
@@ -146,8 +147,13 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 		}
 	}
 
-	os.Setenv("NOTIFY_SOCKET", notifySocket)
-	systemd.SdNotify(true, "READY=1\n")
+	// By default, the server is responsible for notifying systemd
+	// On agent-only nodes, the agent will notify systemd
+	if notifySocket != "" {
+		logrus.Info(version.Program + "-agent is up and running")
```

Comment:
nit: I don't think we don't ever refer to it as `k3s-agent`; it's either just `k3s` or `k3s server`/`k3s agent`
```suggestion
		logrus.Info(version.Program + " agent is up and running")
```